syntax = "proto3";

package stt;

service STT {
  // Unary RPC
  rpc Transcribe (AudioRequest) returns (TranscriptionResponse);

  // Streaming RPC
  rpc TranscribeStream (stream StreamRequest) returns (stream StreamResponse);
}

// ---------- Unary messages ----------
message AudioRequest {
  bytes audio_content = 1;
  string filename = 2;
}

message TranscriptionResponse {
  string text = 1;
  bool success = 2;
  string message = 3;
}

// ---------- Streaming messages ----------
message StreamRequest {
  oneof payload {
    StreamConfig config = 1;
    AudioChunk audio = 2;
  }
}

message StreamConfig {
  string filename = 1;
  string language_code = 2;
  int32 sample_rate = 3;
  bool return_partials = 4;
}

message AudioChunk {
  bytes data = 1;
  bool is_last = 2;
}

message StreamResponse {
  oneof payload {
    PartialTranscript partial = 1;
    FinalTranscript final = 2;
  }
}

message PartialTranscript {
  string text = 1;
  float stability = 2;
}

message FinalTranscript {
  string text = 1;
  bool success = 2;
  string message = 3;
}
